### DISCUSSION TITLE
Jquery Issue with NextJS

### DISCUSSION THREAD

**Discussion by lijoy26** [vercel/next.js#53109] [Help]
### Summary

I was trying to integrate our component library (npm package) build using stencil+storybook+bootstrap.
when I include bootstraps in my library project and try to use it in the nextJS application which is throwing error

I think its jquery include problem, added jquery at library level as well as in the project(nextJS) level still same error. Please help !!

### Additional information

```js
TypeError: Cannot read properties of undefined (reading "jquery")
This error happened while generating the page. Any console logs will be displayed in the terminal window.
Call Stack
Object.jQueryDetection
file:///home/deepthi/projects/go/src/phoenix-portal/node_modules/aui-javaenabled/dist/cjs/bootstrap-b71669f6.js (13491:46)
```

### Example

https://preview.redd.it/frob1wncswdb1.png?width=1360&format=png&auto=webp&s=bbb7ddfd1213abb249b4f3cf9e6d66654aa4e33a

**Comment by Connor-8** [comment]
Are you using the app directory?

**Reply by lijoy26** [reply]
no pages

**Reply by Connor-8** [reply]
My best guess would be this is caused by attempting to server side render components that require the `window` browser api. So ensuring that you've disabled ssr for those components could be a good first step if you haven't tried that already. Here's a link to the documentation on how to do that [https://nextjs.org/docs/pages/building-your-application/optimizing/lazy-loading#with-no-ssr](url). If you have done that already then I'll probably need more info to be able to help further.
